storage2/src/storage.rs

```diff
@@ -0,0 +1,227 @@
+use std::{path::PathBuf, sync::Arc};
+
+use ::metrics::gauge;
+use anyhow::Result;
+use futures::future::BoxFuture;
+use jmt::{
+    storage::{Node, NodeBatch, NodeKey, TreeReader, TreeWriter},
+    WriteOverlay,
+};
+use rocksdb::{Options, DB};
+use tokio::sync::RwLock;
+use tracing::Span;
+
+use penumbra_tct as tct;
+
+use crate::{metrics, State};
+
+#[derive(Clone, Debug)]
+pub struct Storage(Arc<DB>);
+
+impl Storage {
+    pub async fn load(path: PathBuf) -> Result<Self> {
+        let span = Span::current();
+        tokio::task::Builder::new()
+            .name("open_rocksdb")
+            .spawn_blocking(move || {
+                span.in_scope(|| {
+                    tracing::info!(?path, "opening rocksdb");
+                    let mut opts = Options::default();
+                    opts.create_if_missing(true);
+                    opts.create_missing_column_families(true);
+
+                    Ok(Self(Arc::new(DB::open_cf(&opts, path, ["jmt", "nct"])?)))
+                })
+            })
+            .unwrap()
+            .await
+            .unwrap()
+    }
+
+    /// Returns the latest version (block height) of the tree recorded by the
+    /// `Storage`, or `None` if the tree is empty.
+    pub async fn latest_version(&self) -> Result<Option<jmt::Version>> {
+        Ok(self
+            .get_rightmost_leaf()
+            .await?
+            .map(|(node_key, _)| node_key.version()))
+    }
```

Comment:
Assuming we want to maintain the `latest_version` method (why? what do we need it for?), we shouldn't fill in an implementation yet.